src/mono/mono/eglib/garray.c

```diff
@@ -229,7 +229,7 @@ g_array_set_size (GArray *array, gint length)
 	if (length == priv->capacity)
 		return; // nothing to be done
 
-	if (length > priv->capacity) {
+	if (length > GUINT_TO_INT(priv->capacity)) {
```

Comment:
this seems backwards.   two lines above we assert that `length` is positive.  so `(guint) length` will convert without loss of information, unlike converting capacity to signed.

Reply:
You are right. Fixed.